lib/StripeResource.js

```diff
@@ -341,16 +341,25 @@ StripeResource.prototype = {
 
       // If this is a POST and we allow multiple retries, set a idempotency key if one is not
       // already provided.
+      let idempotencyKey;
       if (method === 'POST' && this._stripe.getMaxNetworkRetries() > 0) {
-        if (!headers.hasOwnProperty('Idempotency-Key')) {
-          headers['Idempotency-Key'] = uuid();
+        if (
+          !Object.keys(headers).find((header) => {
+            if (header.toLowerCase() === 'idempotency-key') {
+              idempotencyKey = headers[header];
+              return true;
+            }
+            return false;
+          })
+        ) {
+          headers['Idempotency-Key'] = idempotencyKey = uuid();
```

Comment:
If we're doing this for `Idempotency-Key`, then we should do the same for `Stripe-Version` and `Stripe-Account` https://github.com/stripe/stripe-node/blob/a5885d7447c57eddee2195286b1fd70218feb43e/lib/utils.js#L141-L146

Can you move this functionality to a helper function in `utils` and reuse it for the version and account headers?

Reply:
Here we are checking `params`, not `headers`, that's not the same and may stay as it.

Reply:
There is no problem with setter of headers, only getter of headers must be case insensitive.

Reply:
What I see weird here, is that you are using hardcoded string like this at setters and then lowecased hardcoded strings at getters:
https://github.com/stripe/stripe-node/blob/a5885d7447c57eddee2195286b1fd70218feb43e/lib/StripeResource.js#L127
It looks like a bad style and human errors prone. I would prefer move all private headers to a `http-headers.js` file as lowercase constants and use constants everywhere. Are you OK with that?

Reply:
While I don't think a separate constants file is necessary, I think that's fine!

Reply:
I do think we should have a `getHeader(header, 'Header-Name')` function in utils, and use that to check whether idempotency-key is already defined, and also to get the stripe-version/stripe-account values.

Reply:
(Alternatively, we could normalize the headers hash to be all lower-case keys prior to this processing)

Reply:
> While I don't think a separate constants file is necessary, I think that's fine!

You already have around 10 private HTTP headers sprinkled all around SDK. Some of them are prefixed with `X-Stripe-Client-Telemetry`, some doesn't like `Stripe-Signature` or `Stripe-Version`. Having them all in the same file will set good conventions how to add new one and enforce constants use to prevent any errors in future version regarding them. If you are OK, I'll try to do that today in this PR.

Reply:
Sure, if you think it would help readability to users, please do!